### PR TITLE
apps sc: upgrade harbor to v2.9.1

### LINKED ISSUE #1859: Upgrade harbor
**What Chart should be upgraded?**: https://github.com/goharbor/harbor

Check what the latest version is and upgrade. (Should be v2.9.1 or higher)

**Why should this Chart be upgraded?**:

- [x] We need to perform a security upgrade.
- [ ] We need it for a new feature: *which feature (link to blocked issue)*.
- [ ] We need to keep up to prevent future issues.
- [ ] Other: *Write the reason here*

**Acceptance criteria**:

- I checked the migration of the new Chart:
  - [ ] I upgraded a Chart and determined that no migration steps are needed.
  - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).
- [ ] I tested the functionality provided by the new Chart (e.g., Auth flow, Grafana dashboards, etc.)

### PR DESCRIPTION
**Pull request by lunkan93**
> [!warning]
> **This is public repository, ensure not to disclose:**
> - [x] personal data beyond what is necessary for interacting with this pull request
> - [x] business confidential information, such as customer names

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [x] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [x] kind/admin-change  
- [x] kind/dev-change    
- [ ] kind/security      
- [ ] [kind/adr]()       

### Platform Administrator notice
As of Harbor v2.9, only PostgreSQL 12 or later is supported for external databases. Before upgrading, you should make sure that your external databases are using a supported version of PostgreSQL.

### Application Developer notice
As of Harbor v2.9, Notary V1 is **removed**. If you rely on this for artifact signing, you will need to migrate to one of the alternatives. You can read more about this [here](https://github.com/goharbor/harbor/wiki/Harbor-Deprecates-Notary-v1-Support-in-v2.9.0).

### What does this PR do / why do we need this PR?

Upgraded Harbor to v2.9.1

- Fixes #1859 

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [ ] The change is transparent
  - [ ] The change is disruptive
  - [ ] The change requires no migration steps
  - [x] The change requires migration steps
- Metrics checks:
  - [x] The metrics are still exposed and present in Grafana after the change
  - [x] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [x] The logs do not show any errors after the change
- Network Policy checks:
  - [x] Any changed pod is covered by Network Policies
  - [x] The change does not cause any dropped packages in the `NetworkPolicy Dashboard`
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [x] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

File: config/config/sc-config.yaml [hunk 1]
```diff
@@ -295,38 +293,6 @@ harbor:
...
-    resources:
-      requests:
-        cpu: 20m
-        memory: 125Mi
-      limits:
-        cpu: 250m
-        memory: 250Mi
     affinity:
       podAntiAffinity:
         requiredDuringSchedulingIgnoredDuringExecution:
```
[the remark this answers is not in the document]
Comment (lunkan93) [reply config/config/sc-config.yaml]: Good catch, mistake on my part

File: migration/v0.35/apply/50-harbor.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  execute)
+    if [[ "${CK8S_CLUSTER}" =~ ^(sc|both)$ ]]; then
+      log_info "Upgrading harbor"
+      helmfile_upgrade sc 'app=harbor'
+      log_info "Deleting deprecated notary secrets"
+      kubectl_do sc delete secret -n harbor harbor-notary-cert harbor-notary-ingress-cert --ignore-not-found=true
+    fi
+    ;;
+  rollback)
+    log_warn "rollback not implemented"
```
Comment (OlleLarsson) [inline migration/v0.35/apply/50-harbor.sh:19]:
```suggestion
    log_warn "rollback not applicable"
```

File: migration/v0.35/README.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+    ```bash
+    ./bin/ck8s init sc
+    # or
+    ./migration/v0.35/prepare/50-init.sh
+
+    # check if the netpol IPs need to be updated
+    ./bin/ck8s update-ips sc dry-run
+    # if you agree with the changes apply
+    ./bin/ck8s update-ips sc apply
```
Comment (viktor-f) [inline migration/v0.35/README.md:123]: I would want a migration script that removes any notary config that might have been added to the override config.

**Review by viktor-f — APPROVED** [review]
LGTM

**Review by crssnd — APPROVED** [review]
didn't test it, but looks ok

**Pull request merged**